src/backend/distributed/connection/remote_commands.c

```diff
@@ -1042,6 +1042,12 @@ SendCancelationRequest(MultiConnection *connection)
 {
 	char errorBuffer[ERROR_BUFFER_SIZE] = { 0 };
 
+	if (!PQisBusy(connection->pgConn))
+	{
+		/* no statement in progress, nothing to cancel */
```

Comment:
would be useful to have something `DEBUG4, citus skips sending cancellation as the connection is not running a command` or such for debugging purposes